### PR TITLE
feature/analysis to support sub-graph for TRT engine

### PR DESCRIPTION
**Pull request by Superjomn**
- inference/analysis framework ready for TRT subgraph extraction
- the whole framework runs smoothly from a fluid model as input, and output some predictions
- TODO fix some issues in `executor`, the `TensorRTEngineOp` seems not executed by executor, but the analysis framework works fine, I will work on it in another PR.

### CODE REVIEW

File: paddle/contrib/inference/paddle_inference_api_tensorrt_subgraph_engine.cc [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    executor_->CreateVariables(
+        *inference_program_, sub_scope_ ? sub_scope_ : scope_.get(), 0);
+
+    // Get the feed_target_names and fetch_target_names
+    feed_target_names_ = inference_program_->GetFeedTargetNames();
+    fetch_target_names_ = inference_program_->GetFetchTargetNames();
+    return true;
+  }
+
+  framework::proto::ProgramDesc ProcessProgramDesc(
```
Comment (panyx0718) [inline paddle/contrib/inference/paddle_inference_api_tensorrt_subgraph_engine.cc]: where is this used?

File: paddle/fluid/inference/analysis/analyzer_tester.cc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#include "paddle/fluid/inference/analysis/analyzer.h"
+#include "paddle/fluid/inference/analysis/ut_helper.h"
+
+namespace paddle {
+namespace inference {
+namespace analysis {
+
+TEST_F(DFG_Tester, main) {
+  Analyzer analyser;
+  analyser.Run(&argument);
```
Comment (panyx0718) [inline paddle/fluid/inference/analysis/analyzer_tester.cc:24]: where does argument come from?

File: paddle/fluid/inference/analysis/tensorrt_subgraph_node_mark_pass_tester.cc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#include "paddle/fluid/inference/analysis/ut_helper.h"
+
+namespace paddle {
+namespace inference {
+namespace analysis {
+
+TEST_F(DFG_Tester, tensorrt_subgraph_node_mark_pass) {
+  // init
+  FluidToDataFlowGraphPass pass;
+  ASSERT_TRUE(pass.Initialize(&argument));
```
Comment (panyx0718) [inline paddle/fluid/inference/analysis/tensorrt_subgraph_node_mark_pass_tester.cc:28]: why is the argument from?

File: paddle/fluid/inference/analysis/analyzer_tester.cc (hunk 2 above)
Comment (Superjomn) [reply paddle/fluid/inference/analysis/analyzer_tester.cc:24]: The `argument` is from `DFG_Tester`

File: paddle/contrib/inference/paddle_inference_api_tensorrt_subgraph_engine.cc (hunk 1 above)
Comment (Superjomn) [reply paddle/contrib/inference/paddle_inference_api_tensorrt_subgraph_engine.cc]: cleaned

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**